### ISSUE TITLE
Generate text output from DNN file 

### ISSUE THREAD

**Issue by amitkumartiwari29** [Microsoft/CNTK#887]
Hi,

I am working with the CNTK toolkit and trying to convert audio (.wav) file to Text file (spoken audio to text). I have successfully generated the DNN model files and *.dnn.__AllNodes__.txt using the DumpNodeInfo command. But this text file has no text in it. What do I need to do further to generate actual spoken word as Text?

**Comment by chrisbasoglu** [comment]
Can you descripbe more of your pipleine.  Speech Recognition requires more than just CNTK.  Are you interested in a pre-trained CNTK model to do speech recognition that is already hosted as a REST API?  If so, take a look at [Microsoft Cognitive Servicies](https://www.microsoft.com/cognitive-services/en-us/speech-api).

**Comment by amitkumartiwari29** [comment]
We want to convert all the voice data of our CSRs (customer care calls) to text. For that we want to use the CNTK toolkit.

**Comment by chrisbasoglu** [comment]
Do you have all the other portions of a Spech Recognition pipeline in place? The neural network (CNTK) is only one part of that pipeline.  If you want to  convert all the voice data of your customer care calls to text, and you do not have those pieces of the pipeline in place, you can use Microsoft Cognitive Services to do so.

**Comment by amitkumartiwari29** [comment]
I did not understand "All other portions". I just have CNTK toolkit. I generated the DNN files and its All Nodes Text which have just numbers. Am I missing anything here?

**Comment by dongyu888** [comment]
Speech recognition involves many components, including a pronunciation dictionary, language model and decoding graph, front end signal processing components, and acoustic model. CNTK only helps you to build the acoustic model part. Even for that, you still need to feed in your training data with labels generated using other tools to train the CNTK model. 

 

Based on your description, I think what you have generated is just a randomly initialized CNTK model. What you have dumped is just the model parameters of that initial CNTK model. 

 

Building a speech recognition engine requires years of knowledge and tons of training data. If you don’t have these, the best solution is to call existing speech recognition services like cognitive service Chris pointed out.

**Comment by amitkumartiwari29** [comment]
Thank you so much for the information.

**Comment by n17s** [comment]
Closing. Feel free to reopen.
